VstsTasks/AzureDtlCreateVM/task-funcs.ps1

```diff
@@ -0,0 +1,180 @@
+        {
+            $TemplateParameters = "-labName '$($Lab.Name)' $TemplateParameters"
+        }
+        $null = @(
+            Write-Host "Invoking deployment with the following parameters:"
+            Write-Host "  DeploymentName = $deploymentName"
+            Write-Host "  ResourceGroupName = $resourceGroupName"
+            Write-Host "  TemplateFile = $templateFile"
+            Write-Host "  TemplateParameters = $TemplateParameters"
+        )
+    }
+    else
+    {
+        throw "Unable to locate template file '$TemplateName'. Make sure the template file exists or the path is correctly specified."
+    }
+
+    return Invoke-Expression -Command "New-AzureRmResourceGroupDeployment -Name `"$deploymentName`" -ResourceGroupName `"$resourceGroupName`" -TemplateFile `"$templateFile`" $TemplateParameters"
+}
+
+function Get-AzureDtlLab
+{
+    [CmdletBinding()]
+    param(
+        [string] $LabId
+    )
+
+    $null = @(
+        $labParts = $LabId.Split('/')
+        $labName = $labParts.Get($labParts.Length - 1)
+        Write-Host "Fetching lab '$labName'"
+    )
+
+    return Get-AzureRmResource -ResourceId "$LabId"
+}
+
+function Get-TemplateParameterValue
+{
+    [CmdletBinding()]
+    Param(
+        [string]$Parameters,
+        [string]$ParameterName
+    )
+
+    $pattern = '\-(?<k>\w+)\s+(?<v>\''.*?\''|\$\(.*\)?|\(.*\)?)'
+
+    $value = [regex]::Matches($Parameters, $pattern) | % { if ($_.Groups[1].Value -eq $ParameterName) { return $_.Groups[2].Value } }
+    if ($value)
+    {
+        $value = $value.Trim("'")
+    }
+    
+    return $value
+}
+
+function Validate-TemplateParameters
+{
+    [CmdletBinding()]
+    Param(
+        [string] $Parameters
+    )
+
+    $defaultValues = @{
+        NewVMName = '<Enter VM Name>'
+        UserName = '<Enter User Name>'
+        Password = '<Enter User Password>'
+    }
+
+    $vmName = Get-TemplateParameterValue -Parameters $Parameters -ParameterName 'newVMName'
+    $userName = Get-TemplateParameterValue -Parameters $Parameters -ParameterName 'userName'
+    $password = Get-TemplateParameterValue -Parameters $Parameters -ParameterName 'password'
+
+    $mustReplaceDefaults = $false
+    if ($vmName -and $vmName.Contains($defaultValues.NewVMName))
+    {
+        Write-Host 'WARNING: -newVMName value should be replaced with non-default.'
+        $mustReplaceDefaults = $true
+    }
+    if ($userName -and $userName.Contains($defaultValues.UserName))
+    {
+        Write-Host 'WARNING: -userName value should be replaced with non-default.'
+        $mustReplaceDefaults = $true
+    }
+    if ($password -and $password.Contains($defaultValues.Password))
+    {
+        Write-Host 'WARNING: -password value should be replaced with non-default.'
+        $mustReplaceDefaults = $true
+    }
+
+    if ($mustReplaceDefaults)
+    {
+        throw 'Default values must be replaced. Please review the Template Parameters and modify as needed.'
+    }
+}
+
+function Validate-VMName

```

Comment:
We let the RP return with the error, so as to minimize duplicate checks that may change over time and would therefore make it difficult to maintain this code.